### PR TITLE
[editing-visual] add mixed-pitch

### PR DESCRIPTION
**Pull request by thanhvg**
Add this https://gitlab.com/jabranham/mixed-pitch

### CODE REVIEW

File: layers/+emacs/org/README.org [hunk 1]
```diff
@@ -569,6 +570,9 @@ variable =org-enable-asciidoc-support= to =t=.
     '((org :variables org-enable-asciidoc-support t)))
 #+END_SRC
 
+** Mixed pitched support
+Set the variable =org-enable-mixed-pitch= to =t=.
```
Comment (lebensterben) [inline layers/+emacs/org/README.org:574]: Please also add a warning/note about the interaction with company.
See `Interaction with company` section on its repository.

File: layers/+lang/asciidoc/README.org [hunk 2]
```diff
@@ -22,6 +23,9 @@ To use this configuration layer, add it to your =~/.spacemacs=. You will need to
 add =asciidoc= to the existing =dotspacemacs-configuration-layers= list in this
 file.
 
+** Mixed pitched support
```
Comment (lebensterben) [inline layers/+lang/asciidoc/README.org:26]: Please also add a warning/note about the interaction with company.
See `Interaction with company` section on its repository.

File: layers/+lang/markdown/README.org [hunk 3]
```diff
@@ -107,6 +108,8 @@ list of `("language" "mode")`:
 bunch of languages. Consider opening a PR on the Spacemacs repository to add
 new languages instead of overriding the variable in your dotfile.
 
+** Mixed pitched support
```
Comment (lebensterben) [inline layers/+lang/markdown/README.org:111]: Please also add a warning/note about the interaction with company.
See `Interaction with company` section on its repository.

File: layers/+spacemacs/spacemacs-editing-visual/README.org [hunk 4]
```diff
@@ -17,3 +18,24 @@ edited line in Spacemacs.
 - Highlighting of different indentations
 - Automatic highlighting of numbers
 - Automatic highlighting of parentheses
+* Mixed pitch fonts support
```
Comment (lebensterben) [inline layers/+spacemacs/spacemacs-editing-visual/README.org:21]: Please also add a warning/note about the interaction with company.
See `Interaction with company` section on its repository.

**Review by lebensterben — CHANGES_REQUESTED** [review]
LGTM. All requests are about documentation.

**Review by lebensterben — CHANGES_REQUESTED** [review]
@thanhvg 

After more considerations, I think it's better to do the following instead:

- Make a new layer in `fonts` category with this single package.
- Make a layer variable for example `mixed-pitch-enabled-modes` and defaults to `'(org markdown asciidoc)`
- In the `packages.org` of the layer, for each mode in `mixed-pitch-enabled-modes`, add a hook.
- Also don't forget to update the CHANGELOG

**Comment by lebensterben** [comment]
@thanhvg 
Please review the changes I made.

This should retained the same functionality but I moved the addition to `spacemacs-editing-visual` to a new layer `mixed-pitch`.

And there's no changes to `markdown` or `asciidoc` layers.

I will add you to co-author once I squash the commits and merge them.

**Comment by thanhvg** [comment]
Sorry I'm not having much free time for the last few weeks. I am not a big fan of one package layer. It's convoluted going though all the layer boilerplate to have it in while it can be enabled and support everything inside `dotspacemacs/user-config`. For mixed pitch I think it fits my individual taste more than the community's so I 'd rather close this PR.

One more thing you guys should deprecate CHANGELOG too and enforce better git commit messages instead.

**Comment by lebensterben** [comment]
@thanhvg 
I also don't like one package layer. But we don't currently have any layer configuration variable for any `spacemacs-*` layers.

Because by default these `spacemacs-*` layers are implicitly enabled when you choose the `spacemacs` distribution. Unless the user chose `spacemacs-base` distribution, he won't be able to specify a layer config variable.

**Pull request closed without merging**